TMDBApp/TMDBApp/Requests.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  Requests.swift
+//  TMDBApp
+//
+//  Created by Richard Leh on 19/12/2016.
+//  Copyright © 2016 Richard Leh. All rights reserved.
+//
+
+import Foundation
+import Alamofire
+
+// MARK: API TMDB MOVIE
+private struct TMDBMovieAPI {
+    
+    static let key = "28190e24b6d3d0b73a9b5842bee9d8bf" // 1f54bd990f1cdfb230adb312546d765d​
+    
+    static let upcomingPath = "https://api.themoviedb.org/3/movie/upcoming"
+    static let genrerPath   = "https://api.themoviedb.org/3/genre/movie/list"
+    static let searchPath   = "http://api.themoviedb.org/3/search/movie"
+    static let imagePath    = "https://image.tmdb.org/t/p/w780"
+    
+}
+
+enum RequestError: Error {
+    case InvalidRequest(String)
+}
+
+class Requests: NSObject{
+    
+    class func sharedInstance() -> Requests {
+        struct Singleton {
```

Comment:
I don't need to modify the variable on app's lifetime and I don't need to build this object on the app load, this is .

I'm not so sure about the impacts but I think one of them can be the use excessive of memory.